### PR TITLE
feat(operation-id-naming-convention): add new operation-id-naming-convention rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
'operation-id-naming-convention' rule, which replaces
the legacy 'operation_id_naming_convention' rule.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [x] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/ruleset/src/functions/operation-id-naming-convention.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+const pickBy = require('lodash/pickBy');
```
Comment (padamstx) [inline packages/ruleset/src/functions/operation-id-naming-convention.js:1]: This file is essentially the logic from the legacy rule implementation, with really just cosmetic differences to comply with the spectral rule pattern.

**Review by dpopp07 — APPROVED** [review]
Looks good! I especially love the alternate operation IDs in the unit tests 😂

**Pull request merged**